pkg/discoverymanager/foreign.go

```diff
@@ -136,8 +141,8 @@ func (discovery *Controller) createOrUpdate(data *discoveryData,
 	return nil
 }
 
-func (discovery *Controller) createForeign(
-	ctx context.Context, data *discoveryData,
+func createForeign(
+	ctx context.Context, cl client.Client, scheme *runtime.Scheme, data *discoveryData,
```

Comment:
You should be able to get the scheme from the client, to avoid passing both as parameters (same above).